version: "3.8"

services:
#  next-client:
#    container_name: next-client
#    restart: "on-failure"
#    build:
#      context: ./next-homepage
#    ports:
#      - 3000:3000
  nginx-proxy:
    container_name: nginx-proxy   
    build:
      context: ./nginx
    ports:
      - 80:80
      - 443:443
  
  #  volumes:
  #    - "./next:/usr/src/next"
  #    - "/usr/src/next/node_modules"
  #    - "/usr/src/next/.next" 
  # nginx:
  #   container_name: nginx-proxy   
  #   build:
  #     context: ./nginx
  #   restart: unless-stopped
  #   volumes:
  #     - ./data/nginx:/etc/nginx
  #     - ./data/certbot/conf:/etc/letsencrypt
  #     - ./data/certbot/www:/var/www/certbot
  #   ports:
  #     - "80:80"
  #     - "443:443"
  #   command: "/bin/sh -c 'while :; do sleep 6h & wait $${!}; nginx -s reload; done & nginx -g \"daemon off;\"'"
  # certbot:
  #   image: certbot/certbot
  #   restart: unless-stopped
  #   volumes:
  #     - ./data/certbot/conf:/etc/letsencrypt
  #     - ./data/certbot/www:/var/www/certbot
  #   entrypoint: "/bin/sh -c 'trap exit TERM; while :; do certbot renew; sleep 12h & wait $${!}; done;'"
